src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -3546,6 +3544,20 @@ void *SWITCH_THREAD_FUNC sofia_profile_thread_run(switch_thread_t *thread, void
 		}
 	}
 
+	/* Do gateway cleanups */
+	sofia_glue_del_every_gateway(profile);
+	sofia_reg_check_gateway(profile, switch_epoch_time_now(NULL));
+	sofia_sub_check_gateway(profile, switch_epoch_time_now(NULL));
+	sofia_glue_fire_events(profile);
+
+	switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_NOTICE, "Waiting for worker thread\n");
+
+	if (worker_thread) {
+		switch_thread_join(&st, worker_thread);
+	}
+	else {
+		switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_ERROR, "ERROR: Sofia worker thead failed to start\n");
```

Comment:
That log message may be misleading, should not it be rather something like "Nothing to join here" or similar?